### ISSUE TITLE
1.1.5.+ Program Crashes on Startup

### ISSUE THREAD

**Issue by JunkiEDM** [Nornec/Midinous-Issues#37]
Fresh install, Windows 11 22624.1755
Midi device setup seems to have been successful, not sure what else I can add.
Logs:
[log_error.txt](https://github.com/Nornec/Midinous-Issues/files/11566779/log_error.txt)
[log_full.txt](https://github.com/Nornec/Midinous-Issues/files/11566780/log_full.txt)
[log.txt](https://github.com/Nornec/Midinous-Issues/files/11566782/log.txt)

**Comment by Nornec** [comment]
[settings.zip](https://github.com/Nornec/Midinous-Issues/files/11567135/settings.zip)
Please put the settings.json file which is inside this .zip into your %AppData%/Local/Midinous/user/ directory. Try launching again. If that doesn't work. Please view this thread on Steam Discussions: https://steamcommunity.com/app/1727420/discussions/0/3757723993427180489/?tscn=1685029512

**Comment by JunkiEDM** [comment]
Thanks, this got rid of some of the errors (the old settings.json only had the unquoted version number), but not all of them, I still get this: 
```
>>>>>>>>2023-145.15.49.17 Midinous session startup commencing. This is an error log.
>>>>>>>>2023-145.15.49.17 Release - 1.1.5.4
2023-145.15.49.17 Error occurred attempting to initialize main Midinous app.
2023-145.15.49.17 Exception caught: SteamApi_Init returned false. Steam isn't running, couldn't find Steam, AppId is ureleased, Don't own AppId.
```
[log_error.txt](https://github.com/Nornec/Midinous-Issues/files/11568361/log_error.txt)
[log_full.txt](https://github.com/Nornec/Midinous-Issues/files/11568362/log_full.txt)
[log.txt](https://github.com/Nornec/Midinous-Issues/files/11568363/log.txt)
Error dump from Event Viewer:
[midinous-error-dump-3.csv](https://github.com/Nornec/Midinous-Issues/files/11568390/midinous-error-dump-3.csv)

Hopefully this helps at all for debugging, if needed I can send my local `ucrtbase.dll`, I'll try running it in a Windows Sandbox VM and see if I get the same errors

**Comment by JunkiEDM** [comment]
It launches fine in a Windows Sandbox VM, although the `settings.json` still only includes the unquoted version number until the program is closed so the errors about it are probably unrelated. I compared SHA1 hashes of `ucrtbase.dll` using 7zip on host and vm and they both returned the same hash (`B090EB18F6FB3A96C2F9898E5F6F95A6021D368B`)
Here's the `log_error.txt` contents from the *successful* VM launch:
```
>>>>>>>>2023-145.16.11.48 Midinous session startup commencing. This is an error log.
>>>>>>>>2023-145.16.11.48 Release - 1.1.5.4
2023-145.16.11.48 Error occurred attempting to initialize main Midinous app.
2023-145.16.11.48 Exception caught: SteamApi_Init returned false. Steam isn't running, couldn't find Steam, AppId is ureleased, Don't own AppId.
2023-145.16.11.51 Error while doing a file operation on C:\Users\WDAGUtilityAccount\AppData\Local/Midinous/user/settings.json
2023-145.16.11.51 Exception caught: Input string '1.1.5.4' is not a valid number. Path '', line 1, position 7.
2023-145.16.11.52 Could not read data from C:\Users\WDAGUtilityAccount\AppData\Local/Midinous/user/settings.json
2023-145.16.11.52 Exception caught: Input string '1.1.5.4' is not a valid number. Path '', line 1, position 7.
```

**Comment by Nornec** [comment]
This is fantastic information. Thank you very much! I think i know what's going on now.

I will have the settings file fix in the next hotfix. Thank you for troubleshooting that! As for the .dll issues i think i know what is happening and it may be caused from remnants of .dlls that the program doesn't need to include anymore but did anyway, leading to old versions of .dlls (like crtbase.dll) in the install package, overwriting the need for the program to consume the up to date ones from SYSTEM.

So, i have activated and enabled the 'experimental' branch which you can access through the betas menu (if you own the program). It has some dependency changes and .dll changes which i think may solve the issue (it also includes the settings.json fix by adding quotes to the first line of the settings file). Please try to open and test Midinous using the experimental branch and report back here.

**Comment by Nornec** [comment]
Solved in 1.1.5.5 -- this was due to an issue with PortAudio initializing ASIO devices. Which ASIO devices I'm uncertain, but a fellow user suggested rebuilding portaudio.dll without ASIO support and this worked. I haven't had reports of the app crashing at startup since the update for this specific reason.
